genomic_data_service/parser.py

```diff
@@ -1,12 +1,16 @@
 from asyncio.log import logger
 import abc
-from genomic_data_service.gene_name_lookup import gene_name_lookup
+import pickle
 
 class Parser:
-    def __init__(self, reader, cols_for_index={}, file_path=None):
+    def __init__(self, reader, cols_for_index={}, file_path=None, gene_lookup=False):
         self.reader = reader
         self.cols_for_index = cols_for_index
-        self.file_path = file_path 
+        self.file_path = file_path
+        if gene_lookup:
+            with open("gene_lookup.pickle", "rb") as file:
+                self.gene_symbol_dict = pickle.load(file)
```

Comment:
In general performing actions in `__init__` isn't ideal. It makes instantiating and testing a class more difficult, especially with remote file/network calls. I would think about passing in already loaded data, or another layer of indirection that can read from the file when called for the first time, or push this functionality into its own method.